crates/polars-arrow/src/array/boolean/mod.rs

```diff
@@ -375,6 +375,59 @@ impl BooleanArray {
             validity,
         }
     }
+
+    /// Calculate the number of unique non-null elements in [`Self`]
+    pub fn distinct_count(&self) -> usize {
```

Comment:
I think this should be in `polars-compute` and then that crate should be exposed to `polars-parquet`.

On the fast-path for non-null data. I think we can use the cached null_count values on the underlying bits directly.